cardano-crypto-tests/src/Test/Crypto/EllipticCurve.hs

```diff
@@ -132,6 +138,12 @@ testBLSCurve name _ =
         BLS.blsMult (BLS.blsMult a b) c === BLS.blsMult (BLS.blsMult a c) b
     , testProperty "scalar mult distributive left" $ \(a :: BLS.Point curve) (BigInteger b) (BigInteger c) ->
         BLS.blsMult a (b + c) === BLS.blsAddOrDouble (BLS.blsMult a b) (BLS.blsMult a c)
+    , testProperty "MSM matches naive approach" $ \(psAndSs :: [(BLS.Point curve, BigInteger)]) ->
+        not (null psAndSs) ==>
+        let points = map fst psAndSs
+            scalars = map ((\(BigInteger x) -> x) . snd) psAndSs
+            pairs = NonEmpty.fromList (zip points scalars)
```

Comment:
This can be easily written with list comprehensions:
```suggestion
        let pairs = NonEmpty.fromList [(p, i) | (p, BigInteger i) <- psAndSs]
```